### PR TITLE
Feature run job task, sparkJAR  and SQL task

### PR DESCRIPTION
**Pull request by RajuGujjalapati**
## Description
Added code to support `Run Job, JAR and SQL task` types. Run Job task can trigger another job by its job_name, JAR task will trigger by providing jar file as library and its main class and SQL tasks will be useful to execute SQL objects such as queries, legacy dashboards, files and alerts

## Related Issue
Link to the issue:  [JAR #49 ](https://github.com/Nike-Inc/brickflow/issues/49)
Link to the issue:  [RunJob #55](https://github.com/Nike-Inc/brickflow/issues/55)
Link to the issue: [SQL #119](https://github.com/Nike-Inc/brickflow/issues/119)

## Motivation and Context

1. Databricks now supports creating a "Run Job"  task, which can trigger another job by its job_id, its good to have this feature in brickflow.
2. Databricks now supports creating a "JAR" task. Using the JAR task, you can ensure fast and reliable installation of Java or Scala code in your Databricks jobs.
3. Databricks now supports creating a "SQL" task. Using the SQL task, SQL tasks will be useful to execute SQL objects such as queries, legacy dashboards, files and alerts

## How Has This Been Tested?
Unit tests + deployed test workflow using wheel file

## Screenshots (if appropriate):
1. All test cases passed:
image

2. Tested using brickflow-bundles-test:

image

4. Test workflow with a JAR task created using whl file:
image
5. Test workflow with a Run Job task created using whl file:
run_job
6. Test workflow with a SQL job task created using whl file:
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: brickflow/cli/projects.py [hunk 1]
```diff
@@ -40,7 +40,9 @@
 class BrickflowProject(BaseModel):
     name: str
     path_from_repo_root_to_project_root: str = "."  # used for mono repo
-    path_project_root_to_workflows_dir: str  # used for repos with multiple batches of workflows
+    path_project_root_to_workflows_dir: (
```
Comment (asingamaneni) [inline brickflow/cli/projects.py:43]: What was the reason for this change?
```suggestion
    path_project_root_to_workflows_dir: str
```

File: brickflow/engine/task.py [hunk 2]
```diff
@@ -375,6 +388,203 @@ class NotebookTask(JobsTasksNotebookTask):
...
+
+    Attributes:
+        job_name (str): The name of the job to be run.
+        host (str, optional): The host on which the job is to be run. If not provided, a default host is used.
+        token (str, optional): The token used for authentication. If not provided, a default token is used.
+        job_parameters (Any, optional): The parameters for the job. If not provided, default parameters are used.
+
+    Examples:
+        Below are the different ways in which the RunJobTask class can be used inside workflow (run_job_task).:
+            1. RunJobTask(job_name="dev_object_raw_to_cleansed", host="https://nike.cloud.databricks.com",
```
Comment (asingamaneni) [inline brickflow/engine/task.py:459]:
```suggestion
            1. RunJobTask(job_name="dev_object_raw_to_cleansed", host="YOUR_WORKSPACE_URL",
```

File: brickflow/engine/task.py [hunk 3]
```diff
@@ -375,6 +388,203 @@ class NotebookTask(JobsTasksNotebookTask):
...
+        pause_subscriptions (bool, optional): If true, the alert notifications are not sent to subscribers.
+        subscriptions (dict, optional): The subscriptions for the alert or dashboard.
+        parameters (dict, optional): The parameters for the SQL task.
+        query_id (str, optional): The ID of the SQL query.
+        warehouse_id (str): The ID of the warehouse.
+
+    Examples:
+        Below are the different ways in which the SqlTask class can be used inside workflow (sql_task).:
+            For SQL query: SqlTask(query_id= "4e16dc24-e30d-4683-96d7-cf7da4e263ad", warehouse_id="044a6f42ad7d914a")
+            For SQL file: SqlTask(file_path="/Workspace/Users/raju.gujjalapati@nike.com/raju_sql_test_brickflow.sql",
```
Comment (asingamaneni) [inline brickflow/engine/task.py:505]:
```suggestion
            For SQL file: SqlTask(file_path="/Workspace/Users/YOUR_USERNAME/raju_sql_test_brickflow.sql",
```

File: brickflow/engine/task.py [hunk 4]
```diff
@@ -375,6 +388,203 @@ class NotebookTask(JobsTasksNotebookTask):
...
+              warehouse_id="044a6f42ad7d914a")
```
Comment (asingamaneni) [inline brickflow/engine/task.py:506]: Please remove all the real email id's and id's in the examples

File: brickflow/cli/projects.py (hunk 1 above)
Comment (RajuGujjalapati) [reply brickflow/cli/projects.py:43]: It changed when I ran make fmt command.

File: brickflow/engine/task.py (hunk 4 above)
Comment (RajuGujjalapati) [reply brickflow/engine/task.py:506]: Okay

**Review by asingamaneni — APPROVED** [review]
LGTM. Thanks for your contributions!

**Pull request merged**